source/common/http/http2/codec_impl.cc

```diff
@@ -747,6 +747,10 @@ ClientConnectionImpl::ClientConnectionImpl(Network::Connection& connection,
 Http::StreamEncoder& ClientConnectionImpl::newStream(StreamDecoder& decoder) {
   StreamImplPtr stream(new ClientStreamImpl(*this, per_stream_buffer_limit_));
   stream->decoder_ = &decoder;
+  // If the connection is currently above the high watermark, make sure to inform the new stream.
+  if (connection_.aboveHighWatermark()) {
```

Comment:
Unfortunately can't - unlike the filter level callbacks, there's only ever one subscriber to the Network::Connection callbacks so the connection doesn't know when new streams are created.

We could move this from here to codec_helper addBacllbacks() to be a bit more parallel if you prefer